guides/concepts/grants.md

```diff
@@ -0,0 +1,248 @@
+
+### Who: A group
+
+A group in the `who` relationship is used like `{ type: 'groups', id: someGroupId }`, in which case the grant will apply to all users who are members of that group.
+
+Cardstack comes with a predefined group, `everyone`. It can be used to grant permissions to any user requesting a resource, whether they're authenticated or not. The following example illustrates its usage:
+
+```javascript
+factory
+  .addResource('grants')
+  .withRelated('who', [{ type: 'groups', id: 'everyone' }])
+  .withRelated(
+    'types',
+    { type: 'content-types', id: 'example-blogs' }
+  )
+  .withAttributes({
+    'may-read-resource': true,
+    'may-read-fields': true
+  });
+```
+
+Whoever, we often want to have more specific groups to restrain permissions. Groups are defined as following:
+```javascript
+
+factory.addResource('groups', 'example-readers').withAttributes({
+  'search-query': {
+    filter: {
+      type: { exact: 'example-users' },
+      permissions: { exact: 'cardstack/example-data:read' }
+    }
+  }
+});
+```
+
+Then we can use our newly defined group in a grant as:
+
+```javascript
+factory
+  .addResource('grants')
+  .withRelated('who', [{ type: 'groups', id: 'example-readers' }])
+  .withRelated(
+    'types',
+    { type: 'content-types', id: 'example-blogs' }
+  )
+  .withAttributes({
+    'may-read-resource': true,
+    'may-read-fields': true
+  });
+```
+
+To assign a user to a group we set the group permissions on the user when we define it as:
+
+```javascript
+
+factory.addResource('example-users', 'test-user')
+  .withAttributes({
+    'name': 'Carl Stack',
+    'email-address': 'user@cardstack.com',
+    'permissions': [
+      'cardstack/example-data:read',
+    ]
+  });
+```
+
+### Who: A field
+
+A field defined as `{ type: 'fields', id: someFieldName }`, in which case the set of users will be determined dynamically by looking at the value of given resource's field. The field itself must be either:
+- a relationship to one or many users.
+- the `id` field, which is treated implicitly as a relationship
+to the record itself, meaning you can write a grant against the `id` field in order to give a user permissions to their own user record.
+
+In general, it is better to use groups rather than make lots of separate grants for each user. The biggest use for individual user grants is when you use them via field indirection, so that a resource can contain its own list of owners, etc.
+
+For example, you can have a resource like this that stores its own collaborators list:
+```javascript
+{
+  type: 'posts',
+  id: '1',
+  relationships: {
+    collaborators: {
+      data: [ { type: 'users', id: '1'}, { type: 'users', id: '2' } ]
+    }
+  }
+}
+```
+And write a grant like this that allows people to edit only posts which list them as collaborators, and only if they are in the `unbanned-users` group.
```

Comment:
Wait... what? 😂 I feel like this `unbanned-users` thing came out of nowhere and wasn't explained 🙈 

I would love to have a quick example of how to build a group called `unbanned-users` using the query semantics introduced in the section above.

Reply:
This was from the original comment Ed had, I haven't actually checked if this really works like this and need some specifics. I'll make a little experiment and update accordingly.

Reply:
I clarified this example, please check it again to see if it makese sense.

Reply:
looks good 🎉